### PR TITLE
feat(config): add full enterprise provider

### PR DESCRIPTION
**Pull request by lukesteensen**
This gets us to a working ~~(spike-quality)~~ enterprise provider. It's still ~~mostly spike-quality and therefore~~ feature-flagged off for the time being.

/cc @Zenithar: ~~note that all of the TUF/Uptane verification logic is intentionally left out. Now that we have a working proof of concept, I will begin to fill all of that in.~~

### CODE REVIEW

**Comment by lukesteensen** [comment]
This is now "complete" (i.e. no longer a spike) and ready for folks to review. The one missing bit is that I haven't committed the initial root metadata, since I want to double check that we want to do that 😄 (we require them to be configured in the agent, but I would like to commit and embed them if reasonable).

File: lib/datadog/remote-config/examples/fetch.rs [hunk 1]
```diff
@@ -0,0 +1,36 @@
+use anyhow::{Context, Result};
+use remote_config::{Client, Config};
+
+#[tokio::main]
+async fn main() -> Result<()> {
+    let api_key = std::env::var("DD_API_KEY").context("missing DD_API_KEY env var")?;
+    let app_key = std::env::var("DD_APP_KEY").context("missing DD_APP_KEY env var")?;
```
Comment (spencergilbert) [inline lib/datadog/remote-config/examples/fetch.rs:7]: Does this separately need an application key? IIRC that requirement has been deprecated recently (generally).

File: lib/datadog/remote-config/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,481 @@
...
+    director_client: TUFClient,
+}
+
+impl Client {
+    pub async fn initialize(config: Config) -> Result<Self> {
+        let Config {
+            site,
+            api_key,
+            app_key,
+            hostname,
```
Comment (spencergilbert) [inline lib/datadog/remote-config/src/lib.rs:72]: Curious around the requirements for this, and how the hostname will be provided/obtained.

File: lib/datadog/remote-config/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,481 @@
...
+        let id = Alphanumeric.sample_string(&mut rand::thread_rng(), 21);
+
+        let active_client = proto::Client {
+            state: Some(client_state),
+            id,
+            is_agent: false,
+            client_agent: Some(client_agent),
+            last_seen: std::time::SystemTime::now()
+                .duration_since(std::time::SystemTime::UNIX_EPOCH)
+                .expect("unix epoch is in the past")
```
Comment (spencergilbert) [inline lib/datadog/remote-config/src/lib.rs:110]: Nit? As that's what we _expect_.
```suggestion
                .expect("unix epoch not to be in the past")
```

File: lib/datadog/remote-config/src/lib.rs [hunk 4]
```diff
@@ -0,0 +1,481 @@
...
+            backend_client_state: self.backend_client_state.clone(),
+            ..Default::default()
+        };
+
+        let all_products = self.products.union(&self.new_products);
+        self.active_client.products = all_products.cloned().collect();
+
+        self.active_client.last_seen = std::time::SystemTime::now()
+            .duration_since(std::time::SystemTime::UNIX_EPOCH)
+            .expect("unix epoch is in the past")
```
Comment (spencergilbert) [inline lib/datadog/remote-config/src/lib.rs:255]: Ditto my above nit/question.

**Comment by lukesteensen** [comment]
Closing as this will be moved to a different binary.

**Pull request closed without merging**